### PR TITLE
add Readiness Liveness probes for dataplane components

### PR DESCRIPTION
**Pull request by tharindu1st**

### CODE REVIEW

File: adapter/Dockerfile [hunk 1]
```diff
@@ -46,11 +46,20 @@ RUN \
     && chown -R ${APK_USER}:${APK_USER_GROUP} ${APK_USER_HOME} \
     && echo '[ ! -z "${TERM}" -a -r /etc/motd ] && cat /etc/motd' >> /etc/bash.bashrc; echo "${MOTD}" > /etc/motd
 
+RUN \
+    wget -q https://github.com/grpc-ecosystem/grpc-health-probe/releases/download/v0.4.16/grpc_health_probe-linux-${TARGETARCH} \
```
Comment (Amila-Rukshan) [inline adapter/Dockerfile:50]: shall we add a checksum verification step to confirm the downloaded binary is fine?

File: adapter/resources/check_health.sh [hunk 2]
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+# --------------------------------------------------------------------
+# Copyright (c) 2021, WSO2 Inc. (http://wso2.com) All Rights Reserved.
```
Comment (AmaliMatharaarachchi) [inline adapter/resources/check_health.sh:3]:
```suggestion
# Copyright (c) 2023, WSO2 LLC. (http://wso2.com) All Rights Reserved.
```

**Review by CrowleyRajapakse — APPROVED** [review]

File: helm-charts/templates/data-plane/gateway-components/gateway-runtime/gateway-runtime-deployment.yaml [hunk 3]
```diff
@@ -85,6 +87,20 @@ spec:
...
             {{- end }}
+            - name: enforcer-keystore-secret-volume
+              mountPath: /home/wso2/security/truststore/apk.crt
+            {{- if and .Values.wso2.apk.dp.gatewayRuntime.deployment.enforcer.configs .Values.wso2.apk.dp.gatewayRuntime.deployment.enforcer.configs.tls }}
+              subPath: {{ .Values.wso2.apk.dp.gatewayRuntime.deployment.enforcer.configs.tls.caCertFileName | default "ca.crt" }}
+            {{- else }}
+              subPath: ca.crt
+            {{- end }}
+            - name: enforcer-keystore-secret-volume
+              mountPath: /home/wso2/security/truststore/enforcer.crt
```
Comment (AmaliMatharaarachchi) [inline helm-charts/templates/data-plane/gateway-components/gateway-runtime/gateway-runtime-deployment.yaml:98]: why do we add the enforcer public cert to it's own truststore?

File: management-server/resources/check_health.sh [hunk 4]
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+# --------------------------------------------------------------------
+# Copyright (c) 2021, WSO2 Inc. (http://wso2.com) All Rights Reserved.
```
Comment (AmaliMatharaarachchi) [inline management-server/resources/check_health.sh:3]:
```suggestion
# Copyright (c) 2023, WSO2 LLC. (http://wso2.com) All Rights Reserved.
```

File: helm-charts/templates/data-plane/gateway-components/gateway-runtime/gateway-runtime-deployment.yaml (hunk 3 above)
Comment (tharindu1st) [reply helm-charts/templates/data-plane/gateway-components/gateway-runtime/gateway-runtime-deployment.yaml:98]: this is needed for mtls in probe

**Review by AmaliMatharaarachchi — APPROVED** [review]

**Pull request merged**